test/test_torch.py

```diff
@@ -7447,75 +7311,44 @@ def test_cholesky_batched_many_batches(self):
-                          (-1.56, 4.00, -8.67, 1.75, 2.86),
-                          (9.81, -4.09, -4.57, -8.61, 8.99))).t()
-
-        # make sure 'a' is symmetric PSD
-        a = torch.mm(a, a.t())
-        a, b = cast(a), cast(b)
-
-        # upper Triangular Test
-        U = torch.cholesky(a, True)
-        x = torch.cholesky_solve(b, U, True)
-        self.assertLessEqual(b.dist(torch.mm(a, x)), 1e-12)
-
-        # lower Triangular Test
-        L = torch.cholesky(a, False)
-        x = torch.cholesky_solve(b, L, False)
-        self.assertLessEqual(b.dist(torch.mm(a, x)), 1e-12)
-
-        # default arg Test
-        L_def = torch.cholesky(a)
-        x_def = torch.cholesky_solve(b, L_def)
-        self.assertLessEqual(b.dist(torch.mm(a, x_def)), 1e-12)
+        from common_utils import cholesky_solve_test_helper
+        for (k, n), upper in product(zip([2, 3, 5], [3, 5, 7]), [True, False]):
+            b, A, L = cholesky_solve_test_helper((n,), (n, k), cast, upper)
+            x = torch.cholesky_solve(b, L, upper=upper)
+            b_ = torch.matmul(A, x)
+            self.assertEqual(b_, b)
 
     @skipIfNoLapack
     def test_cholesky_solve(self):
         self._test_cholesky_solve(self, lambda t: t)
 
     @staticmethod
     def _test_cholesky_solve_batched(self, cast):
-        from common_utils import random_symmetric_pd_matrix
+        from common_utils import cholesky_solve_test_helper
 
-        def cholesky_solve_test_helper(A_dims, b_dims, cast, upper):
-            A = cast(random_symmetric_pd_matrix(*A_dims))
-            L = torch.cholesky(A, upper)
-            b = cast(torch.randn(*b_dims))
-            return A, L, b
-
-        for upper in [True, False]:
-            # test against cholesky_solve: one batch with both choices of upper
-            A, L, b = cholesky_solve_test_helper((5, 1), (1, 5, 10), cast, upper)
-            x_exp = torch.cholesky_solve(b.squeeze(0), L.squeeze(0), upper=upper)
-            x = torch.cholesky_solve(b, L, upper=upper)
-            self.assertEqual(x, x_exp.unsqueeze(0))
-
-            # test against cholesky_solve in a loop: four batches with both choices of upper
-            A, L, b = cholesky_solve_test_helper((5, 4), (4, 5, 10), cast, upper)
+        def cholesky_solve_batch_helper(A_dims, b_dims, cast, upper):
+            b, A, L = cholesky_solve_test_helper(A_dims, b_dims, cast, upper)
             x_exp_list = []
-            for i in range(4):
-                x_exp = torch.cholesky_solve(b[i], L[i], upper=upper)
-                x_exp_list.append(x_exp)
-            x_exp = torch.stack(x_exp_list)
+            for i in range(b_dims[0]):
+                x_exp_list.append(torch.cholesky_solve(b[i], L[i], upper=upper))
+            x_exp = torch.stack(x_exp_list)  # Stacked output
+            x_act = torch.cholesky_solve(b, L, upper=upper)  # Actual output
+            self.assertEqual(x_exp, x_act)  # Equality check
```

Comment:
nit: switch the order of the args in assertEqual

Reply:
Sure, thank you.